cli/commands/preview/update.ts

```diff
@@ -0,0 +1,82 @@
+import os from 'node:os';
+import path from 'node:path';
+import { __, _n } from '@wordpress/i18n';
+import { uploadArchive, waitForSiteReady } from 'cli/lib/api';
+import { createArchive } from 'cli/lib/archive';
+import { getAuthToken } from 'cli/lib/auth';
+import { addPreviewSiteToAppdata, getSnapshotsFromAppdata } from 'cli/lib/snapshots';
+import { validateSiteFolder } from 'cli/lib/validation';
+import { Logger, LoggerError } from 'cli/logger';
+import { RegisterCommand, OutputFormat } from 'cli/types';
+
+enum LoggerAction {
+	VALIDATE = 'validate',
+	ARCHIVE = 'archive',
+	UPLOAD = 'upload',
+	READY = 'ready',
+	APPDATA = 'appdata',
+}
+
+async function runCommand(
```

Comment:
The logic in this function is very similar to that in `cli/commands/preview/create.ts`. The difference is only really that we pass a third argument to `uploadArchive`.